test/lib/server/services/eosReport/EosReportService.test.js

```diff
@@ -109,15 +109,21 @@ module.exports = () => {
     });
 
     it ('should create a new EoS report autofilled with previous EoS report', async () => {
-        const past = new Date() - SHIFT_DURATION;
+        const now = Date.now();
+        const future = new Date(now + SHIFT_DURATION);
         const info = 'Important information\nfor the next tester\ncontaining #punctuation...';
-        const request = customTimeECSEosReportRequest(past, info);
+        const request = customTimeECSEosReportRequest(now, info);
 
         const log = await eosReportService.createLogEntry(ShiftTypes.ECS, request, { userId: 1 });
-        expect(log.text).to.equal(formattedCustomTimeECSEosReport(past, info));
+        expect(log.text).to.equal(formattedCustomTimeECSEosReport(now, info));
 
-        const response = await shiftService.getShiftData({ userId: 1 }, ShiftTypes.ECS);
-        expect(response.previousInfo).to.equal(info);
+        const response = await shiftService.getShiftData({ userId: 1 }, ShiftTypes.ECS, future);
+        expect(response.previousInfo.value).to.equal(info);
+    });
+
+    it ('should throw an error in previousInfo if no previous EoS report is found', async () => {
+        const response = await shiftService.getShiftData({ userId: 1 }, ShiftTypes.SLIMOS, Date.now());
+        expect(response.previousInfo.errorMessage).to.equal('No SLIMOS EoS reports found from the previous shift');
     });
```

Comment:
You are changing shift service, so your tests must be about ShiftService. 
You do not need any customEoS report object, you only need to:
- create a log 8 hours ago with a fake EoS report (containing only the correct title and a body that match your regex)
- call your UserService function 

Do the same but on the frontend side

Reply:
Simplified ShiftService tests